src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10945,29 +10911,31 @@ private ExpressionSyntax ParseTermWithoutPostfix(Precedence precedence)
                         return this.AddError(this.CreateMissingIdentifierName(), ErrorCode.ERR_InvalidExprTerm, this.CurrentToken.Text);
                     }
                 case SyntaxKind.IdentifierToken:
-                    if (this.IsTrueIdentifier())
                     {
```

Comment:
braces are to prevent some conflicts with pattern name clases in a switch statement.